src/test/scala/retryFunction/RetryFunctionSpec.scala

```diff
@@ -1,33 +1,50 @@
 package retryFunction
 
+import org.scalatest.GivenWhenThen
 import org.scalatest.flatspec.AnyFlatSpec
 import org.scalatest.matchers.should.Matchers
 
-class RetryFunctionSpec extends AnyFlatSpec with Matchers {
+class RetryFunctionSpec extends AnyFlatSpec with Matchers with GivenWhenThen{
 
   "RetryFunction.retry" should "return '3' without additional retries" in {
+    Given("input block function '() => 1 + 2' and accept function 'res => res > 0'. Number of potential retries: 3 (100, 1000, 2000)")
+    When("retry function is executed")
+    Then("result of execution should be 3")
     RetryFunction.retry[Int](
       block = () => 1 + 2,
       accept = res => res > 0,
       retries = List(100, 1000, 2000)
     ) should equal(3)
   }
 
-  it should "do only 1 retry" in {
+  it should "do only 1 retry if accept function return true after it" in {
+    Given("util variable of Utility type. Only one method tempIncrement is used in this test - for incrementation of initial temp value")
+    Given("input block function '() => util.tempIncrement()' and accept function 'res => res == 1'. Number of potential retries: 3 (100, 100, 100)")
+
     val util = Utility(-1)
+
+    When("retry function is executed")
+
     RetryFunction.retry[Int](
       block = () => util.tempIncrement(),
       accept = res => res == 1,
       retries = List(100, 100, 100)
     )
 
-    //checking that temp variable was increased only 1 time
+    Then("util.temp value should be equal 1 (increase only 1 time)")
+
     util.temp should equal(1)
   }
 
-  it should "do all retries and return -1" in {
+  it should "return result of block function if it's not match accept function after all retries" in {
+    Given("util variable of Utility type. Only one method tempIncrement is used in this test - for incrementation of initial temp value")
+    Given("input block function '() => {\n        util.tempIncrement()\n        -1\n      }' and accept function 'res => res > 0'. Number of potential retries: 3 (100, 200, 300)")
+
     val util = Utility(-1)
 
+    When("retry function is executed")
```

Comment:
Invocation of `retry` function should be after `When` and before `Then` function.
After `Then` should be only assertions.

Please, see for more details: https://martinfowler.com/bliki/GivenWhenThen.html

Reply:
Code was refactored. Now retry function is executed after When and before Then. Result of this execution is assigned to retryResult variable. This variable is used for assertions after Then